tests/api-diff/modified_variant_type_tests.t

```diff
@@ -0,0 +1,261 @@
+Here we generate a `.mli` file with a variant type:
+
+  $ cat > ref.mli << EOF
+  > type rank = Ace | King | Queen | Number of int
+  > EOF
+
+We generate the .cmi file
+
+  $ ocamlc ref.mli
+
+# Tests for different kind of modifications to a variant type:
+
+### Adding a constructor to a variant type:
+
+  $ cat > add_constructor.mli << EOF
+  > type rank = Ace | King | Queen | Jack | Number of int
+  > EOF
+
+We generate the .cmi file
+
+  $ ocamlc add_constructor.mli 
+
+Run the api-watcher on the two cmi files
+
+  $ api-diff ref.cmi add_constructor.cmi
+  diff module Add_constructor:
+   type rank =
+    ...
+  + | Jack
+  
+  [1]
+
+### Removing a constructor from a variant type:
+
+  $ cat > remove_constructor.mli << EOF
+  > type rank = Ace | King | Queen
+  > EOF
+
+We generate the .cmi file
+
+  $ ocamlc remove_constructor.mli
+
+Run the api-watcher on the two cmi files
+
+  $ api-diff ref.cmi remove_constructor.cmi
+  diff module Remove_constructor:
+   type rank =
+    ...
+  - | Number of int
+  
+  [1]
+
+### Modifying a constructor's arguments in a variant type: 
+
+  $ cat > modify_constructor_type.mli << EOF
+  > type rank = Ace | King | Queen | Number of float
+  > EOF
+
+We generate the .cmi file
+
+  $ ocamlc modify_constructor_type.mli
+
+Run the api-watcher on the two cmi files
+
+  $ api-diff ref.cmi modify_constructor_type.cmi
+  diff module Modify_constructor_type:
+   type rank =
+    ...
+  - | Number of (
+  -    int
+  +    float
+    )
```

Comment:
The `-` before the constructor name should be removed!
Also, I find the parentheses a bit weird
(And they might interfere with the OCaml syntax. For instance,
```
type t1 = A of int * int
type t2 = A of (int * int)
```
are valid OCaml syntax which mean different things: one has the tuple "unboxed" while the other has it boxed: see `utop -dparsetree` !)